freetextresponse/private/view.js

```diff
@@ -4,46 +4,125 @@ function FreeTextResponseView(runtime, element) {
             }),
             success: function buttonSubmitOnSuccess(response) {
-                textareaParent.removeClass();
-                textareaParent.addClass(userInputClass);
-                textareaParent.addClass(response.indicator_class);
                 usedAttemptsFeedback.text(response.used_attempts_feedback);
-                submitParent.removeClass();
-                submitParent.addClass(response.submit_class);
+                //submitParent.removeClass();
+                //submitParent.addClass(response.submit_class);
                 problemProgress.text(response.problem_progress);
-                wordCountError.text(response.word_count_message);
                 submissionReceivedMessage.text(response.submitted_message);
                 buttonSubmit.text('Submit');
+                userAlertMessage.text(response.user_alert);
+                //saveParent.removeClass();
+                //saveParent.addClass(response.submit_class);
+                
+                // Update Indicators
+                _setIndicators(response.indicator_class, response.indicator_visibility_class);
 
-                runtime.notify('submit', {
-                    state: 'end'
-                });
+                //runtime.notify('submit', {
+                //    state: 'end'
+                //});
             },
             error: function buttonSubmitOnError() {
                 runtime.notify('error', {});
             }
         });
         return false;
     });
+
+    buttonSave.on('click', function () {
+        buttonSave.text('Checking...');
+        //runtime.notify('save', {
+        //    message: 'Saving...',
+        //    state: 'start'
+        //});
+        $.ajax(urlSave, {
+            type: 'POST',
+            data: JSON.stringify({
+                'student_answer': $element.find('.student_answer').val()
+            }),
+            success: function buttonSaveOnSuccess(response) {
+                textareaParent.removeClass();
+                textareaParent.addClass(userInputClass);
+                textareaParent.addClass(response.indicator_class);
+                usedAttemptsFeedback.text(response.used_attempts_feedback);
+                problemProgress.text(response.problem_progress);
+                submissionReceivedMessage.text(response.submitted_message);
+                buttonSave.text('Save');
+                userAlertMessage.text(response.user_alert);
+                
+                // Update Indicators
+                _setIndicators(response.indicator_class, response.indicator_visibility_class);
+                
+                //runtime.notify('save', {
+                //    state: 'end'
+                //});
+            },
+            error: function buttonSaveOnError() {
+                runtime.notify('error', {});
+            }
+        });
+        return false;
+    });
+
+    textareaParent.on('keydown', function() {
```

Comment:
Nit: Always put a space after `function` (you did this above).
Also on the consistency note, any reason you're not naming this anonymous function,
whereas you are doing so above?